core/codegen/src/lib.rs

```diff
@@ -59,7 +59,7 @@
 //! ```
 
 #[macro_use] extern crate quote;
-extern crate proc_macro;
+use proc_macro;
```

Comment:
This change is not strictly necessary (since `extern crate` will continue to be allowed at least in the 2018 edition), but it *does* increase the minimum nightly `rustc`. For that reason I think it should be done in a separate commit if at all - for reference, it was done on the async branch in f0c3177 and ca65d88.

Reply:
Thanks for the explanation and the kind advice !
I've split commit.